backend/src/Edna/Library/DB/Query.hs

```diff
@@ -87,7 +98,25 @@ targetsWithProjects targetSqlId = runSelectReturningList' $ select $
         tTargetId t ==. val_ (SqlSerial targetId)
       Nothing -> \_ -> val_ True
 
-getCompoundById :: SqlId 'CompoundId -> Edna (Maybe CompoundRec)
+getTargetByName :: Text -> Edna (Maybe TargetRec)
+getTargetByName name = runSelectReturningOne' $ select $ do
+  targets <- all_ $ esTarget ednaSchema
+  guard_ (LDB.tName targets ==. val_ name)
+  pure targets
+
+insertTarget :: Text -> Edna (Maybe (Text, Word32))
```

Comment:
1. Why did you change it to return `Maybe`? It must always return `Just` and it's responsibility of _this_ function to return `Just`. The contract of this function is that it always inserts the target or returns ID of the existing one. This function knows that `getTargetByName` must return `Just` because it sets `onConflict` as necessary. The caller currently just assumes that this function always returns `Just` and uses `error` (aka "impossible happened"). If we remove `Maybe`, the caller won't need to think whether `Nothing` is possible.
2. Pairing ID with name sounds like something the DB layer doesn't need to care about. This layer doesn't know anything about API that we expose, and if we forget this knowledge it will be strange to return `(name, ID)` pair. Returning only ID is ok because it's auto-generated. Returning the whole object is also ok. But returning some combination of fields sounds like something the Service layer should take care of.

Reply:
Also, since this function is now exported, it should have a comment about this detail. Here the argument "it's obvious what this function does" doesn't work because inserting a target is not trivial because duplicate names are not allowed. So we should say that in case of a conflict this function does nothing and simply returns the existing ID.

And in general please try to explain what you are doing. Usually I see almost no comments, no PR description, no commit messages.

Reply:
>Pairing ID with name sounds like something the DB layer doesn't need to care about.

DB layer should construct types which are used by Service layer (Domain values) and Service layer should only work with these Domain types and get/pass them from/to API or DB layer. But as I already said it is not a panacea and for example for Library module we would have to write Domain types and two converters: from/to DB types and from/to API types. Instead of this we just skip them and use only API and DB types.

So, I just want to say that splitting our logic on three layers is already good enough and juggling with types looks a bit over engineering here (I do not suggest to refuse them, but to not think a lot about them). But I agree that returning only IDs is better